### PR TITLE
Release 1.1.2 - Country Risk Frontend

### PR DESCRIPTION
**Pull request by alexsilva-CGI**
Hello,

This merge is relative to the next release, with all the fixes and changes documented in the changelog.

Thanks,
Alexandre Silva

### CODE REVIEW

**Comment by SebastianBezold** [comment]
I've created the 3rd party IP issues via Eclipse dash license tool. The following issues where created: 

- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7952
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7953
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7954
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7955
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7956
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7957
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7958
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7959
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7960
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7961
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7962
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7963
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7964
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7965
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7966
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7967
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7968
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7969

**Comment by SebastianBezold** [comment]
After the last commit there was a version bump for terser. 
IP issue for the new version: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/7952

File: charts/country-risk/country-risk-frontend/templates/_helpers.tpl
Comment (hzierer) [inline charts/country-risk/country-risk-frontend/templates/_helpers.tpl:1]: Please add a proper copyright + license header
{{/*
...
*/}}

File: charts/country-risk/values.yaml
Comment (hzierer) [inline charts/country-risk/values.yaml:36]: are you sure this will work this way?

File: charts/helper/backend/_helpers.tpl
Comment (hzierer) [inline charts/helper/backend/_helpers.tpl:1]: Please add a proper copyright + license header

File: charts/helper/frontend/_helpers.tpl
Comment (hzierer) [inline charts/helper/frontend/_helpers.tpl:1]: Please add a proper copyright + license header

File: sonar-project.properties
Comment (hzierer) [inline sonar-project.properties:1]: Is there not also sonarcloud available in Tractus-X?

**Review by hzierer — CHANGES_REQUESTED** [review]
Please check my comments, I tried to mark duplicate issues too

File: sonar-project.properties
Comment (alexsilva-CGI) [reply sonar-project.properties:1]: Hi @hzierer. Yeah, I believe we only had sonar available for catena-x. Is it also necessary for Tractus-X I suppose?

Comment (hzierer) [reply sonar-project.properties:1]: Not strictly necessary (at least for QG4) but should make lives easier in the future as Tractus-X is your leading repo and Catena-X is only a fork

File: charts/country-risk/README.md
Comment (hzierer) [inline charts/country-risk/README.md:1]: I meant of course our "official" [Tractus-X repo](https://eclipse-tractusx.github.io/charts)
(or [our dockerhub](https://hub.docker.com/u/tractusx) as soon as you have moved there)

so currently it should be:
```
helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
helm install <your_release_name> tractusx-dev/country-risk-frontend --namespace <your_namespace>
```

File: charts/country-risk/country-risk-backend/README.md
Comment (hzierer) [inline charts/country-risk/country-risk-backend/README.md:1]: see my comment on other README.md above

**Review by hzierer — CHANGES_REQUESTED** [review]

File: README.md
Comment (hzierer) [inline README.md:1]: Can you please enhance the README.md: a description about Country Risk itself + its purpose.
A bit like in the [backend repo](https://github.com/eclipse-tractusx/vas-country-risk-backend/blob/main/README.md).
Especially since frontend is the new leading repo. Maybe move the long description here and reference to it in the backend only

**Review by hzierer — CHANGES_REQUESTED** [review]

File: README.md
Comment (alexsilva-CGI) [reply README.md:1]: I added some more information regarding the project, and also an reference to our docs folder

Comment (hzierer) [reply README.md:1]: Hi @alexsilva-CGI,
I opened a [PR](https://github.com/catenax-ng/tx-vas-country-risk-frontend/pull/2) to your fork with some suggestions for this Readme. Was faster to create the PR than describing it here.

Cheers Harry

File: charts/country-risk/country-risk-backend/README.md
Comment (carslen) [reply charts/country-risk/country-risk-backend/README.md:27]: In Line 13/14 you fixed it, but forgot this one

File: charts/country-risk/values.yaml
Comment (hzierer) [inline charts/country-risk/values.yaml:180]: Please see comment line 36

Comment (alexsilva-CGI) [reply charts/country-risk/values.yaml]: Yep, these values are for ingress. In regards to the new value is the way I changed fine?

**Review by hzierer — APPROVED** [review]
LGTM

File: charts/country-risk/values.yaml
Comment (carslen) [inline charts/country-risk/values.yaml:53]: Is this a orphaned value? Did not find it in any Helm template file being used...

If it's orphaned, please remove section also in sub chart `values.yaml` files.

File: charts/country-risk/values.yaml
Comment (carslen) [inline charts/country-risk/values.yaml:200]: Is this a orphaned value? Did not find it in any Helm template file being used...

If it's orphaned, please remove section also in sub chart `values.yaml` files.

**Review by carslen — CHANGES_REQUESTED** [review]
Please refer to comments.

**Comment by SebastianBezold** [comment]
Not sure, what is happening here, but there are a lot of new unchecked 3rd party dependencies. I created the IP issues to check the 3rd party libraries. Please track them and update the `DEPENDENCIES` ones they are resolved. Please also try to avoid making this PR bigger and bigger. It is already at a size, that now requires a manual IP check for project content

__Project content review__: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8427

__List of 3rd party library checks__: 

- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8386
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8387
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8388
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8374
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8389
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8390
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8391
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8392
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8393
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8394
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8395
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8396
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8397
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8398
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8399
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8400
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8401
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8402
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8403
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8404
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8375
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8405
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8406
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8407
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8408
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8409
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8410
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8411
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8412
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8413
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8414
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8415
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8416
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8417
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8418
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8419
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8420
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8421
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8422
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8423
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8424
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8425

File: charts/country-risk/values.yaml
Comment (fabiodmota) [reply charts/country-risk/values.yaml:345]: this was just a simply default password so when no profile is passed it can connect the microservice to the postgres, in other value envs it will get them from vault

File: charts/country-risk/country-risk-backend/templates/secret-pgadmin4.yaml
Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/templates/secret-pgadmin4.yaml:25]: > Not sure, what is happening here, but there are a lot of new unchecked 3rd party dependencies. I created the IP issues to check the 3rd party libraries. Please track them and update the `DEPENDENCIES` ones they are resolved. Please also try to avoid making this PR bigger and bigger. It is already at a size, that now requires a manual IP check for project content
> 
> **Project content review**: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8427
> 
> **List of 3rd party library checks**:
> 
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8386
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8387
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8388
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8374
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8389
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8390
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8391
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8392
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8393
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8394
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8395
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8396
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8397
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8398
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8399
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8400
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8401
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8402
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8403
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8404
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8375
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8405
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8406
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8407
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8408
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8409
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8410
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8411
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8412
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8413
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8414
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8415
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8416
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8417
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8418
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8419
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8420
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8421
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8422
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8423
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8424
> * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8425

Thanks Sebastian , I updated and changed some libs that were being found with transient vulnerability's .  To be able to pass this security findings  https://jira.catena-x.net/browse/CXRM-1042 and since the parent lib didn't had new releases i needed to fixed the sub libs and upgrade them.

Comment (SebastianBezold) [reply charts/country-risk/country-risk-backend/templates/secret-pgadmin4.yaml]: > > Not sure, what is happening here, but there are a lot of new unchecked 3rd party dependencies. I created the IP issues to check the 3rd party libraries. Please track them and update the `DEPENDENCIES` ones they are resolved. Please also try to avoid making this PR bigger and bigger. It is already at a size, that now requires a manual IP check for project content
> > **Project content review**: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8427
> > **List of 3rd party library checks**:
> > 
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8386
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8387
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8388
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8374
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8389
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8390
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8391
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8392
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8393
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8394
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8395
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8396
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8397
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8398
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8399
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8400
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8401
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8402
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8403
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8404
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8375
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8405
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8406
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8407
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8408
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8409
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8410
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8411
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8412
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8413
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8414
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8415
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8416
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8417
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8418
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8419
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8420
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8421
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8422
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8423
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8424
> > * https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8425
> 
> Thanks Sebastian , I updated and changed some libs that were being found with transient vulnerability's . To be able to pass this security findings https://jira.catena-x.net/browse/CXRM-1042 and since the parent lib didn't had new releases i needed to fixed the sub libs and upgrade them.

Hi @fabiodmota, thanks for the info. Can you please paste some context from the link you provided here, since it is not accessible for everyone. Updates are fine with me, but still we need to wait for the IP issue approvals

Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/templates/secret-pgadmin4.yaml]: Dependabot detected vulnerability on d3-color lib that is used by d3-zoom lib that is used by react simple maps.
My only used lib is react simple maps that was already on the latest version (3.0.0)
So in order to upgrade the d3-color I upgraded the parent manually by overriding d3-zom

image

With my upgrade on that many libs that the d3-zom uses also updated , and fixed my issue like we can see on the attach.

File: charts/country-risk/country-risk-backend/values.yaml
Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/values.yaml]: done

File: charts/country-risk/country-risk-backend/values.yaml
Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/values.yaml]: done

File: charts/country-risk/country-risk-backend/values.yaml
Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/values.yaml]: yes the email its needed so i provided a default without mention of the consortia

File: charts/country-risk/values.yaml
Comment (fabiodmota) [reply charts/country-risk/values.yaml]: done like above

File: charts/country-risk/values.yaml
Comment (fabiodmota) [reply charts/country-risk/values.yaml]: done

File: Dockerfile
Comment (fabiodmota) [reply Dockerfile]: done

File: charts/country-risk/values.yaml
Comment (fabiodmota) [reply charts/country-risk/values.yaml]: done

File: charts/country-risk/values.yaml
Comment (fabiodmota) [reply charts/country-risk/values.yaml]: done

**Review by hzierer — CHANGES_REQUESTED** [review]

File: charts/country-risk/country-risk-backend/values.yaml
Comment (fabiodmota) [reply charts/country-risk/country-risk-backend/values.yaml:45]: done

**Review by carslen — APPROVED** [review]
LGTM

IP Issues still under EF review. GH Release will be created after all IP issues/checks have been approved.

**Pull request merged**
